app/controllers/api_controller/collection_config.rb

```diff
@@ -0,0 +1,110 @@
+class ApiController
+  class CollectionConfig < Config::Options
```

Comment:
Sounds good.

There is still one or two list like accessors left for `collection_config[]`. Let's see if I can avoid that.